packages/klighd-core/src/proxy-view/proxy-view-actions.ts

```diff
@@ -0,0 +1,117 @@
+import { SendModelContextAction } from "../actions/actions";
+import { DISymbol } from "../di.symbols";
+import { OptionsRegistry } from "../options/options-registry";
+import { RenderOptionsRegistry } from "../options/render-options-registry";
+import { SynthesesRegistry } from "../syntheses/syntheses-registry";
+import { ProxyView } from "./proxy-view";
+
+//////// Actions ////////
+
+/** Wrapper action around {@link SetUIExtensionVisibilityAction} which shows the proxy.
+  * Otherwise the proxy-view would be invisible. */
+export type ShowProxyViewAction = SetUIExtensionVisibilityAction
+
+export namespace ShowProxyViewAction {
+    export function create(): ShowProxyViewAction {
+        return SetUIExtensionVisibilityAction.create({
+            extensionId: ProxyView.ID,
+            visible: true,
+        })
+    }
+}
+
+/** Sent from the {@link ProxyView} to the action handler to avoid Stackoverflows. */
+export interface SendProxyViewAction extends Action {
+    kind: typeof SendProxyViewAction.KIND
+    proxyView: ProxyView
+}
+
+export namespace SendProxyViewAction {
+    export const KIND = 'sendProxyViewAction'
+
+    export function create(proxyView: ProxyView): SendProxyViewAction {
+        return {
+            kind: KIND,
+            proxyView
+        }
+    }
+}
+
+/**
+ * Handles {@link SendProxyViewAction}s to get the {@link ProxyView} instance
+ * aswell as {@link SendModelContextAction}s to redirect the content to the {@link ProxyView}.
+ */
+@injectable()
+export class ProxyViewActionHandler implements IActionHandler, IActionHandlerInitializer {
+    private proxyView: ProxyView;
+    @inject(DISymbol.SynthesesRegistry) private synthesesRegistry: SynthesesRegistry;
+    @inject(DISymbol.RenderOptionsRegistry) private renderOptionsRegistry: RenderOptionsRegistry;
+    @inject(DISymbol.OptionsRegistry) private optionsRegistry: OptionsRegistry;
+    private onChangeRegistered: boolean;
+
+    handle(action: Action): void | Action | ICommand {
+        if (action.kind === SendProxyViewAction.KIND) {
+            const sPVAction = action as SendProxyViewAction;
+            this.proxyView = sPVAction.proxyView;
+
+            // Register to receive updates on registry changes
+            if (!this.onChangeRegistered) {
+                // Make sure the rendering cache is cleared when the renderings change
+                this.synthesesRegistry.onChange(() => { this.proxyView.clearRenderings(); this.proxyView.clearPositions() });
+                this.optionsRegistry.onChange(() => this.proxyView.clearRenderings());
+                // Make sure to be notified when rendering options are changed
+                this.renderOptionsRegistry.onChange(() => this.proxyView.updateOptions(this.renderOptionsRegistry));
+                this.onChangeRegistered = true;
+            }
+        } else if (action.kind === SendModelContextAction.KIND && this.proxyView !== undefined) {
```

Comment:
for readability, this and the next case should have the same ordering i.e. either check action type first in both cases or check whether proxy view is defined first in both cases